### PR TITLE
Botany Cargo Points

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Intertwines Botany and Cargo and encourages co-operation via seed samples (from TG).

- Allows botany to send seeds to cargo---if the seed is an exotic variety then cargo can ship it to centcomm to earn points based on the rarity of the seed.
 - Once that seed is sent, you cannot send it again until you increase the potency on the seed--meaning, if you send a bluespace tomato seed the first time---you'll earn 50 points. The second time you send it, if the potency of the new seed is *higher* than the first time you sent it, you'll earn the difference in potency as points and potency of the new seed will be the new baseline.

- Plant analyzers have been updated to allow you to see the rarity of seeds.

Related

- Re-organized the MegaSeed vendor so that seeds are listed alphabetically.
- Removes the plastellium seeds from the MegaSeed vendor
 - these are a mutations of fly amanitas, not something that is meant to start in the vendor.

Coder Stuffs

- Moved hydroponics defines to _DEFINES so this actually works (and most defines should be there anyway).

### CODE REVIEW

File: code/modules/hydroponics/seed.dm [hunk 1]
```diff
@@ -47,6 +47,7 @@
 	set_trait(TRAIT_PLANT_ICON,           0)            // Icon to use for the plant growing in the tray.
 	set_trait(TRAIT_PRODUCT_COLOUR,       0)            // Colour to apply to product icon.
 	set_trait(TRAIT_BIOLUM_COLOUR,        0)            // The colour of the plant's radiance.
+	set_trait(TRAIT_RARITY,				  0)			// How rare the plant is. Used for giving points to cargo when shipping off to Centcom.
```
Comment (tigercat2000) [inline code/modules/hydroponics/seed.dm]: pretty sure all the other ones were actually just spaced out to that level, given that github is displaying this wrong

**Comment by Fox-McCloud** [comment]
Oops. forgot peanuts.

**Comment by Fox-McCloud** [comment]
And rice...

**Comment by taukausanake** [comment]
This sounds interesting. As a buy that ONLY plays Botanist I can't say I'll use this feature much. I'll be busy trying to mass up my planty kingdom of ambrosia-looking plants. That won't be my reason for not using this however. Mainly because I'm pretty sure Cargo won't give two shits about 50 more points that will be pretty limited in amount.

I like it though. Can you quickly explain the rarity though?

**Comment by Fox-McCloud** [comment]
rarity is just how rare CentComm considers the plant to be--nothing more, nothing less--the rarer a plant, the more points it gives you when first sent.

**Pull request merged**